main.go

```diff
@@ -0,0 +1,172 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	helmclient "github.com/mittwald/go-helm-client"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	registriesv1alpha1 "github.com/mittwald/harbor-operator/api/v1alpha1"
+	"github.com/mittwald/harbor-operator/controllers"
+	opconfig "github.com/mittwald/harbor-operator/controllers/config"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	// +kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+	log      = logf.Log.WithName("cmd")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	utilruntime.Must(registriesv1alpha1.AddToScheme(scheme))
+	// +kubebuilder:scaffold:scheme
+}
+
+func main() {
+	flag.StringVar(&opconfig.MetricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.BoolVar(&opconfig.EnableLeaderElection, "enable-leader-election", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&opconfig.HelmClientRepoCachePath, "helm-client-repo-cache-path",
+		"/tmp/.helmcache", "helm client repository cache path")
+	flag.StringVar(&opconfig.HelmClientRepoConfPath, "helm-client-repo-conf-path",
+		"/tmp/.helmconfig", "helm client repository config path")
+
+	flag.Parse()
+
+	viper.SetEnvPrefix("HARBOR_OPERATOR")
+
+	replacer := strings.NewReplacer("-", "_")
+
+	viper.SetEnvKeyReplacer(replacer)
+
+	viper.AutomaticEnv()
```

Comment:
This will probably not work, viper doesn't seem to be able to replace `StringVar` Flags with the corresponding env value: https://github.com/spf13/viper/issues/85

The `config` package would have to explicitly call viper.Get() to make this work.